semeio/jobs/rft/gendata_rft.py

```diff
@@ -157,9 +160,12 @@ def run(
 
             # Aggregate dataframe for all wells and report steps.
             dframes.append(trajectory_df)
+        else:
+            failed = True
 
-    if csvfile is not None:
+    if csvfile is not None and dframes:
         pd.concat(dframes, ignore_index=True, sort=False).to_csv(csvfile, index=None)
 
-    with open("GENDATA_RFT.OK", "w") as fh:
-        fh.write("GENDATA RFT completed OK")
+    if dframes and not failed:
```

Comment:
Isnt `if dframe` abduntant? If that check were to ever fail, the job will also be `failed = True`?

Would also be good to add some logging here perhaps, as to why the job has failed? If we can do that, can we also be more specific about why/where it fails?

One option here is to move the writing of the `OK` file one level up, to `main_entry_point`, then this part of the code can just raise errors. Then the function call can be wrapped in a try/except and handle the errors accordingly.

Reply:
The `if dframe` is needed there to cover the scenario where `well_times` is empty.

Reply:
Moved OK file writing up one level.